scripts/lint/markdown-heuristic-lint.py

```
@@ -0,0 +1,320 @@
+    findings: List[str] = []
+
+    # Skip lines inside code blocks.
+    # Use line.strip() for fence detection to handle indented fences correctly.
+    in_code_block = False
+    fence_char: Optional[str] = None
+    code_block_lines: set = set()
+
+    for i, line in enumerate(lines):
+        lstripped = line.strip()
+        if not in_code_block:
+            if lstripped.startswith("```") or lstripped.startswith("~~~"):
```

Ruff (0.15.10)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)